src/include/migraphx/op/pooling.hpp

```diff
@@ -48,6 +48,18 @@ struct pooling
         assert(lengths[0] <= (input.lens()[2] + 2 * padding[0]));
         assert(lengths[1] <= (input.lens()[3] + 2 * padding[1]));
 
+        // for padding mode same (either same_upper or same lower)
+        // output shape is computed as:
+        if(padding_mode == same)
```

Comment:
`compute_shape` shouldn't use padding mode. Instead, the padding should be set so that the output shape will be correct. We do the same thing for convolution.

MIOpen doesn't take padding mode, so if we dont set the correct padding then it can't compute the pooling correctly.

Reply:
fixed accordingly